scripts/dev/snapshot

```diff
@@ -1,7 +1,82 @@
-#! /bin/sh
+#!/bin/sh
+#
+# Create a PHP release snapshot (tar.gz, tar.bz2, tar.xz) from the current Git
+# repository files. Archive files are created in the current php-src directory.
+# Used, for example, to create a nightly PHP release and similar use cases.
+# Beside the bare Git repository files, the snapshot archive also includes
+# generated configure script, configuration headers, PHP parsers, PHP lexers,
+# and similar generated files.
+#
+# SYNOPSIS:
+#   snapshot
+#
+# Environment:
+#   The following environment variables can be used:
+#
+#   DISTNAME  Override the name of the archive. Default=php-snapshot
+#             DISTNAME=php-YYYY-MM-DD snapshot
 
-if test -n "$1"; then
-  flags="DISTNAME=$1"
+DISTNAME=${DISTNAME:-php-snapshot}
+
+# Go to project root directory.
+cd $(CDPATH= cd -- "$(dirname -- "$0")/../../" && pwd -P)
+
+git archive --format=tar --prefix=$DISTNAME/ HEAD | tar xvf -
```

Comment:
I'm not sure I get this line. This is creating the archive without the generated files?

Reply:
yes to be able to include generated files in the archive:
./buildconf --force
./scripts/dev/genfiles
...